test/src/test/java/org/corfudb/integration/LogReplicationRoutingQueueIT.java

```diff
@@ -312,6 +373,11 @@ public void provideFullStateData(LRFullStateReplicationContext context) {
                             .buildPartial();
                     recordId++;
                     context.transmit(message);
+
+                    if (startedTransmitting != null && startedTransmitting.getCount() > 0) {
+                        startedTransmitting.countDown();
```

Comment:
until the transaction is committed on L383, no data will have been written to queue.  I suggest moving this countdown() operation after the commit so that we can be sure that data from the old snapshot was written before the sync was cancelled.